core/server/services/stats/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Stats
+ * A collection of utilities for handling stats of the site
+ */
+const debug = require('ghost-ignition').debug('stats:cache');
+const moment = require('moment');
+const models = require('../../models');
+const StatsCache = require('./cache');
+
+async function initStats() {
+    const result = await models.Post.query(qb => qb.orderBy('created_at', 'ASC').limit(1)).fetchAll({columns: ['created_at']});
+    const createdAt = result.models[0].get('created_at');
+
+    const stats = {
+        total_members: await models.Member.count(),
```

Comment:
This operation might become time consuming, as well as the rest of `.count()` calls, for large datasets. We'd need performance numbers to evaluate if it's worth keeping Bookshelf model call (suspecting it's not) or maybe drop down a level and query directly through knex. 

To start with performance metrics, you would need to populate the db with dummy members (don't have a public script that I could share :grimacing:). Would need to generate perf numbers for sets of 10K, 50K, 100K members.  Then would measure how long it takes to execute `models.Member.count()` for each set.

For example through `consoleTime` (or [hook into](https://spin.atomicobject.com/2017/03/27/timing-queries-knexjs-nodejs/) bookshelf model directly):
```
console.time('memberCount');

await models.Member.count()

console.timeEnd('memberCount');
```

Similar metrics but with lower numbers (5K, 10K, 30K) would need to be done for posts.

These numbers would lead to further decision making :crossed_fingers:

Reply:
Here are the numbers for Members. I've run the query 50 times and calculated the numbers.

| | 10K model | 10K knex |
| - | ---- | ---- |
| Avg | 31.481ms | 26.795ms |
| Min | 20.762ms 2nd | 9.868ms 1st |
| Max | 43.019ms 50th | 40.010ms 50th |

| | 50K model | 50K knex |
| - | ---- | ---- |
| Avg | 39.171ms | 28.613ms |
| Min | 23.665ms 2nd | 11.960ms 1st |
| Max | 57.316ms 50th | 43.915ms 50th |

I haven't checked posts and tags. But I believe the result would be similar. 

According to the test result, I suggest using knex directly is better for performance than using bookshelf models. 

Particularly, the first run of knex is about 2x faster than model.

If you want more numbers, I'll test it and send them to you.

Reply:
I've shared the tool I used to seed 50K data at #12272.